bcm27xx: add support for 6.12 kernel version

graysky2 opened the pull request:
This PR builds on https://github.com/openwrt/openwrt/pull/16547 and https://github.com/namiltd/openwrt/pull/53 which is not yet merged and is an initial draft at including the 6.12 series kernels.

## What fails to build when compiling an image with `CONFIG_BUILDBOT=y`
* Currently, the build will fail without applying patches from @namiltd which are intended to be temporary until mac80211 can be updated for the 6.12 series of kernel (https://github.com/hauke/backports).

## What needs attention
* I removed patches if they provide Github workflows, wireless drivers, and defconfigs, but I may have missed some so a review by others is welcomed.
* I am unsure if I correctly populated the `config-6.12` files.  Feedback and collaboration is welcomed.

```
Build system: x86/64
Build-tested: bcm2712/rpi-5
Run-tested: bcm2712/rpi-5
```
![yes](https://github.com/user-attachments/assets/03ebb5e8-62e1-4538-977e-bee8dd41e846)

luizluca committed:
kernel: modules: netdevices: add realtek DSA modules

Uses upstream DSA switch modules (rtl8365mb, rtl8366), similar to
RTL8367C and rtl8366rb swconfig drivers.

The package dependencies exclude targets built without kernel CONFIG_OF.

It also fixes the rtl8366rb LED support.

Signed-off-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

mhei committed:
mxs: fix image generation for I2SE Duckbills

The standard U-Boot boot scripts for Duckbills expect the
Linux kernel and device tree files installed below /boot
within the (ext4) root filesystem.
Also a raw zImage is expected instead of uImage.

Extend the SD card generation accordingly and while at,
install all possible Duckbill DT blobs there.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

georgemoussalem committed:
qualcommax: add ipq50xx target

Introduce support for the Qualcomm IPQ50xx SoC.
This series adds support for the following components:
- minimal boot support: GCC/pinctrl/watchdog/CPUFreq/SDI (upstreamed)
- USB2 (upstreamed)
- Thermal/Tsens
- PCIe gen2 1&2-lane PHY and controller
- PWM and PWM LED
- QPIC SPI NAND controller
- CMN PLL Block (provider of fixed rate clocks to GCC/ethernet/more.)
- Ethernet: IPQ5018 Internal GE PHY (1 gbps)
- Remoteproc MPD driver for IPQ5018 (2.4G) & QCN6122 (5/6G) Wifi

Co-developed-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
qca-ssdk: add support for ipq50xx

The codename for IPQ50xx is Maple (abbreviated as 'MP'), so let's pass
the codename to allow the QCA-SSDK to build for the IPQ50xx SoC.

In addition, disable compiling the MP_PHY driver in favor of a native
driver being upstreamed.

Co-developed-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
qca-nss-dp: add support for IPQ50xx

Add support for the Qualcomm IPQ50xx in the QCA NSS dataplane driver.
The QCA implementation uses depracated DMA api calls and a downstream SCM
call, so convert to proper Linux DMA and SCM api calls.
In addition, add fixed-link support to support SGMII which is used to
connect the internal IPQ50xx switch to an external switch (ex. QCA8337)

Co-developed-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
ath11k-firmware: add wifi firmware for IPQ5018

Add ability to download and package ath11k firmware for IPQ5018.

As part of commit 172ccf7, the source is pointed to the new QCA repo.
Until QCA publishes updated firmware for IPQ5018 and QCN6122, we need
to download the firmware from the old QUIC repo. As such, add a new
download routine for IPQ5018/QCN6122 to fetch the firmware files from
the old repo.

While at it, add support for packaging BDFs for IPQ5018-based boards
and update iwinfo to recognize IPQ5018 wifi.

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
mac80211: ath11k: Support setting bdf-addr and caldb-addr via DT

IPQ5018 uses different BDF and caldb addresses for vairous boards,
so let's support reading these addresses from the device tree.

Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
ath11k-firmware: add wifi firmware for QCN6122

Add ability to download and package ath11k firmware for QCN6122.
QCN6122 is specific/exclusive to the IPQ5018 platform and firmware
files are publishes in a subdirectory of the IPQ5018 firmware files.

While at it, add support for packaging BDFs for QCN6122 wifi and
update iwinfo to recognize QCN6122 wifi.

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
mac80211: ath11k: add support for QCN6122 wifi

Add QCN6122 platform support.

QCN6122 is a PCIe based solution that is attached to and enumerated
by the WPSS (Wireless Processor SubSystem) Q6 processor.

Though it is a PCIe device, since it is not attached to APSS processor
(Application Processor SubSystem), APSS will be unaware of such a decice
and hence it is registered to the APSS processor as a platform device(AHB).
Because of this hybrid nature, it is called as a hybrid bus device.

As such, QCN6122 is a hybrid bus type device and follows the same codepath
as for WCN6750.

This is a reversed engineered and heavily simplified version of below
downstream patch:
https://git.codelinaro.org/clo/qsdk/oss/system/feeds/wlan-open/-/ \
blob/NHSS.QSDK.12.4.5.r2/mac80211/patches/232-ath11k-qcn6122-support.patch

Co-developed-by: George Moussalem <george.moussalem@outlook.com>
Signed-off-by: Sowmiya Sree Elavalagan <ssreeela@codeaurora.org>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
mac80211: ath11k: fix remapped ce access on 64-bit OS

https://lore.kernel.org/linux-wireless/TYZPR01MB55563B3A689D54D18179E5B4C9192@TYZPR01MB5556.apcprd01.prod.exchangelabs.com/

Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem committed:
qualcommax: ipq50xx: add support for Linksys MX2000 and MX5500

Add support for Linksys MX2000 (Atlas 6) and MX5500 (Atlas 6 Pro).
These devices are completely identical except for the secondary wifi
chip used for 5Ghz: QCN6102 is used on MX2000 while QCN9024 is used
on MX5500

Speficiations:
* SoC: Qualcomm IPQ5018 (64-bit dual-core ARM Cortex-A53 @ 1.0Ghz)
* Memory: Winbond W634GU6NB-11 (512 MiB DDR3-933)
* Serial Port: 3v3 TTL 115200n8
* Wi-Fi: IPQ5018 (2x2 2.4 Ghz 802.11b/g/n/ax)
* Wi-Fi: MX2000: QCN6102 (2x2:2 5 Ghz 802.11an/ac/ax)
         MX5500: QCN9024 (4x4:4 5 Ghz 802.11an/ac/ax)
* Ethernet: IPQ5018 integrated virtual switch connected to an external
            QCA8337 switch (4 Ports 10/100/1000 GBASE-T)
* Flash: Macronix MX35UF2GE4AD (256 MiB)
* LEDs: 1x multi-color PWM LED
* Buttons: 1x WPS (GPIO 27 Active Low)
           1x Reset (GPIO 28 Acive Low)

Flash instructions (in case of MX2000, else replace with MX5500 images):
1. On OEM firmware, login to the device (typically at http://192.168.1.1) and click 'CA'
in the bottom right corner -> Connectivity -> Manual Upgrade. Alternatively, browse to
http://<router IP>/fwupdate.html.
Upgrade firmware using openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin image.
Optionally install on second partition, after first boot check actual partition:
fw_printenv -n boot_part
and install firmware on second partition using command in case of 2:
mtd -r -e kernel -n write openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin kernel
and in case of 1:
mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin alt_kernel
2. Installation using serial connection from OEM firmware (default login: root, password: admin):
fw_printenv -n boot_part
In case of 2:
flash_erase /dev/mtd12 0 0
nandwrite -p /dev/mtd12 openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin
or in case of 1:
flash_erase /dev/mtd14 0 0
nandwrite -p /dev/mtd14 openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin
After first boot install firmware on second partition:
mtd -r -e kernel -n write openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin kernel
or:
mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq50xx-linksys_mx2000-squashfs-factory.bin alt_kernel
3. Back to the OEM firmware.
Download firmware from OEM website:
MX2000: https://support.linksys.com/kb/article/585-en/
MX5500: https://support.linksys.com/kb/article/587-en/
From serial or SSH:
fw_printenv boot_part
in case of 1:
mtd -r -e alt_kernel -n write FW_MX2000_1.1.7.210469_prod.img alt_kernel
else in case of 2:
mtd -r -e kernel -n write FW_MX2000_1.1.7.210469_prod.img kernel

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd committed:
kernel: update drivers to be compatible with kernel 6.12

Update drivers to be compatible with kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel: update packages to be compatible with kernel 6.12

Update kernel packages to be compatible with kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

mhei committed:
mxs: image: slightly adjust whitespace (no functional change)

Seems that the common style is to have whitespace around the operators
and the indent is single tab.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

graysky2 committed:
x86: 6.12: manually rebuilding patches

Manually rebuilding patches:
 - 103-pcengines_apu6_platform.patch

Signed-off-by: John Audia <therealgraysky@proton.me>

namiltd committed:
generic: 6.12: fix compilation warnings

Fix compilation warnings.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
generic: 6.12: fix undefined references

Fix undefined references to kallsyms_token_table and kallsyms_token_index.

Co-authored-by: Aditya Nugraha <vortexilation@gmail.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
pcs-xpcs: 6.12: Add missing depends

Add missing depends kmod-mdio-devres.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
x86: enable 6.12 testing kernel

The x86 now supports 6.12 kernel as testing.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
firewall4: fix reading kernel version

Fix reading kernel version for kernels with revision 0 e.g. 6.12
Repair incorrect shift of the revision number causing incorrect value for > 256.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

graysky2 committed:
kernel: update dependency of kmod-i2c-piix4

Add 6.12 specific dependency

Package kmod-i2c-piix4 is missing dependencies for the following libraries:
i2c-smbus.ko

Signed-off-by: John Audia <therealgraysky@proton.me>

namiltd committed:
kernel: add new kmod and update kmod-tpm

Create a crypto-lib-aescfb package needed for new deps for kmod-tpm

Package kmod-tpm is missing dependencies for the following libraries:
ecdh_generic.ko
kpp.ko
libaescfb.ko

Co-authored-by: John Audia <therealgraysky@proton.me>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

mantas-p committed:
kernel: adjust kmod-phy-aquantia dependencies for v6.12 kernel

Driver moved to different CRC library in v6.12

Signed-off-by: Mantas Pucka <mantas@8devices.com>

namiltd committed:
kernel/9pnet: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
package/fs-ntfs: disable for kernel 6.12

Disabled due to removal from kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/hwmon-max6697: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/i2c-i801: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/iio-bmp280: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel: 6.12: add Intel ethernet libraries

Add Common Intel Ethernet library (libeth) and
Intel Ethernet library (libie) for kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/lan743x: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel: 6.12: add snd hda scodec component module

Add snd hda scodec component module

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/usb-net-rtl8152: fix missing dependency

Fix missing dependency on kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

mhei committed:
mxs: adapt default package list for Duckbill devices

Replace outdated package names, order the remaining.

firewall4 is added again, since it is enabled by default.
The device is not a router by default, so this package
makes no sense in the default config. User can enable it
by theirself, or it will be automatically pulled via
dependency, e.g. luci-app-firewall.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

namiltd committed:
package/fs-fscache: update for kernel 6.12

Update KCONFIG and FILES for kernel 6.12 compatibility

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel: update fb package on x86 for kernel 6.12

For kernel 6.12 the fb package uses video-common.ko instead of fbdev.ko.
The patch switches these files depending on the kernel version.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

graysky2 committed:
kernel: bcm27xx/bcm2712: enable testing kernel

Enable testing kernel for bcm2712

Signed-off-by: John Audia <therealgraysky@proton.me>

namiltd commented:
Do: make kernel_oldconfig

graysky2 commented:
> Do: make kernel_oldconfig

All that did was undo the options I manually added.
```
diff --git a/target/linux/bcm27xx/config-6.12 b/target/linux/bcm27xx/config-6.12
index a4545a3a6957..d55f73bc1edb 100644
--- a/target/linux/bcm27xx/config-6.12
+++ b/target/linux/bcm27xx/config-6.12
@@ -1,5 +1,4 @@
 CONFIG_64BIT=y
-# CONFIG_AF8133J is not set
 CONFIG_ARCH_BINFMT_ELF_EXTRA_PHDRS=y
 CONFIG_ARCH_CORRECT_STACKTRACE_ON_KRETPROBE=y
 CONFIG_ARCH_DEFAULT_KEXEC_IMAGE_VERIFY_SIG=y
@@ -40,18 +39,12 @@ CONFIG_ARM_ARCH_TIMER_EVTSTREAM=y
 CONFIG_ARM_GIC=y
 CONFIG_ARM_GIC_V3=y
 CONFIG_ARM_GIC_V3_ITS=y
-# CONFIG_ARM_NI is not set
 CONFIG_ARM_PSCI_FW=y
 CONFIG_AUDIT_ARCH_COMPAT_GENERIC=y
-# CONFIG_BACKLIGHT_KTD2801 is not set
-# CONFIG_BACKLIGHT_RPI is not set
-# CONFIG_BACKLIGHT_LM3509 is not set
-# CONFIG_BACKLIGHT_MP3309C is not set
 # CONFIG_BASE_SMALL is not set
 # CONFIG_BCACHEFS_FS is not set
 # CONFIG_BCM2712_MIP is not set
 # CONFIG_BRCM_CHAR_DRIVERS is not set
-# CONFIG_BT_HCIUART_AML is not set
 CONFIG_BUILTIN_RETURN_ADDRESS_STRIPS_PAC=y
 CONFIG_CC_HAVE_SHADOW_CALL_STACK=y
 CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG=y
@@ -66,7 +59,6 @@ CONFIG_CPUMASK_OFFSTACK=y
 CONFIG_CPU_BIG_ENDIAN=y
 CONFIG_CPU_MITIGATIONS=y
 CONFIG_CPU_RMAP=y
-# CONFIG_CPUSETS_V1 is not set
 CONFIG_CRYPTO_ECB=y
 CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC=y
 CONFIG_CRYPTO_LIB_GF128MUL=y
@@ -76,34 +68,7 @@ CONFIG_DCACHE_WORD_ACCESS=y
 CONFIG_DEBUG_INFO=y
 CONFIG_DMA_DIRECT_REMAP=y
 CONFIG_DMA_NEED_SYNC=y
-# CONFIG_DRM_DISPLAY_DP_AUX_CEC is not set
-# CONFIG_DRM_DISPLAY_DP_AUX_CHARDEV is not set
-# CONFIG_DRM_PANEL_BOE_TH101MB31UIG002_28A is not set
-# CONFIG_DRM_PANEL_BOE_TV101WUM_LL2 is not set
-# CONFIG_DRM_PANEL_HIMAX_HX83102 is not set
-# CONFIG_DRM_PANEL_HIMAX_HX83112A is not set
-# CONFIG_DRM_PANEL_ILITEK_ILI9805 is not set
-# CONFIG_DRM_PANEL_ILITEK_ILI9806E is not set
-# CONFIG_DRM_PANEL_ILITEK_ILI9882T is not set
-# CONFIG_DRM_PANEL_JDI_LPM102A188A is not set
-# CONFIG_DRM_PANEL_LINCOLNTECH_LCD197 is not set
-# CONFIG_DRM_PANEL_LG_SW43408 is not set
-# CONFIG_DRM_PANEL_NOVATEK_NT36672E is not set
-# CONFIG_DRM_PANEL_RAYDIUM_RM692E5 is not set
-# CONFIG_DRM_PANEL_RAYDIUM_RM69380 is not set
-# CONFIG_DRM_PANEL_SAMSUNG_S6E3FA7 is not set
-# CONFIG_DRM_PANEL_SYNAPTICS_R63353 is not set
-# CONFIG_DRM_PANEL_TPO_Y17P is not set
-# CONFIG_DRM_PANEL_WAVESHARE_TOUCHSCREEN is not set
-# CONFIG_DRM_RP1_DSI is not set
-# CONFIG_DRM_RP1_DPI is not set
-# CONFIG_DRM_RP1_VEC is not set
-# CONFIG_DRM_PANIC is not set
-# CONFIG_DRM_PANTHOR is not set
-# CONFIG_DRM_POWERVR is not set
-# CONFIG_DRM_XE is not set
 CONFIG_DTC=y
-# CONFIG_DWC_PCIE_PMU is not set
 CONFIG_EDAC_SUPPORT=y
 CONFIG_EXCLUSIVE_SYSTEM_RAM=y
 CONFIG_FIX_EARLYCON_MEM=y
@@ -141,7 +106,6 @@ CONFIG_GENERIC_STRNLEN_USER=y
 CONFIG_GENERIC_TIME_VSYSCALL=y
 CONFIG_GPIO_CDEV=y
 # CONFIG_GPIO_FSM is not set
-# CONFIG_GPIO_SLOPPY_LOGIC_ANALYZER is not set
 # CONFIG_GPIO_VIRTUSER is not set
 CONFIG_HARDIRQS_SW_RESEND=y
 CONFIG_HAS_DMA=y
@@ -157,7 +121,6 @@ CONFIG_IRQ_FORCED_THREADING=y
 CONFIG_IRQ_MSI_LIB=y
 CONFIG_IRQ_WORK=y
 # CONFIG_LAN966X_OIC is not set
-# CONFIG_LEDS_LP5569 is not set
 # CONFIG_LEDS_TRIGGER_ACTPWR is not set
 # CONFIG_LEDS_TRIGGER_INPUT is not set
 CONFIG_LIBFDT=y
@@ -165,9 +128,7 @@ CONFIG_LOCK_DEBUGGING_SUPPORT=y
 CONFIG_LOCK_SPIN_ON_OWNER=y
 CONFIG_LRU_GEN_WALKS_MMU=y
 # CONFIG_LWQ_TEST is not set
-# CONFIG_MEDIA_PCI_HAILO is not set
 # CONFIG_MEM_ALLOC_PROFILING is not set
-# CONFIG_MEMCG_V1 is not set
 CONFIG_MIGRATION=y
 CONFIG_MMU_LAZY_TLB_REFCOUNT=y
 CONFIG_MODULES_USE_ELF_RELA=y
@@ -175,7 +136,6 @@ CONFIG_MODULES_USE_ELF_RELA=y
 CONFIG_MUTEX_SPIN_ON_OWNER=y
 CONFIG_NEED_DMA_MAP_STATE=y
 CONFIG_NEED_SG_DMA_LENGTH=y
-# CONFIG_NET_9P_USBG is not set
 # CONFIG_NETKIT is not set
 CONFIG_NET_EGRESS=y
 CONFIG_NET_FLOW_LIMIT=y
@@ -220,9 +180,7 @@ CONFIG_RPS=y
 CONFIG_RWSEM_SPIN_ON_OWNER=y
 CONFIG_SERIAL_8250_FSL=y
 CONFIG_SERIAL_MCTRL_GPIO=y
-# CONFIG_SENSORS_ASUS_ROG_RYUJIN is not set
 CONFIG_SMP=y
-# CONFIG_SND_PIMIDI is not set
 CONFIG_SOCK_RX_QUEUE_MAPPING=y
 CONFIG_SOFTIRQ_ON_OWN_STACK=y
 CONFIG_SPARSEMEM=y
@@ -230,7 +188,6 @@ CONFIG_SPARSEMEM_EXTREME=y
 CONFIG_SPARSEMEM_VMEMMAP=y
 CONFIG_SPARSEMEM_VMEMMAP_ENABLE=y
 CONFIG_SPARSE_IRQ=y
-# CONFIG_SPI_RP2040_GPIO_BRIDGE is not set
 CONFIG_SPLIT_PMD_PTLOCKS=y
 CONFIG_SPLIT_PTE_PTLOCKS=y
 CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU=y
@@ -251,7 +208,6 @@ CONFIG_TREE_SRCU=y
 CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_USER_STACKTRACE_SUPPORT=y
 CONFIG_VDSO_GETRANDOM=y
-# CONFIG_W1_MASTER_UART is not set
 CONFIG_XPS=y
 CONFIG_XZ_DEC_BCJ=y
 CONFIG_ZONE_DMA32=y
-- 
2.47.1
```

namiltd commented:
Please rebase.

graysky2 commented:
Rebased but am wondering if there is a more efficient method.  I did this:
```
git format-patch -6 # to save my 6 commits
git reset --hard upstream/master # this is openwrt master
git fetch nam-6.12
git rebase nam-6.12/6.12 # to apply your stuff
git am 0001-kernel-bcm27xx-Create-kernel-files-for-v6.12-from-v6.patch 0002-kernel-bcm27xx-Restore-kernel-files-for-v6.6.patch 0003-kernel-bcm27xx-bcm2712-enable-testing-kernel.patch 0004-patches-6.12-bcm27xx-refresh-patch-set.patch 0005-bcm27xx-bcm2712-refresh-6.12-kernel-config.patch 0006-bcm27xx-bcm2712-refresh-6.12-kernel-config.patch```
```

namiltd commented:
```
git reset --hard 7c9c26f47490f2ce05765860b9dd4e21d31ef437
git cherry-pick 586bd9fda2b38610d5246f7ece6dc35c19235314~..29408cff28e76c4c9a10d6d6c8f08c470d7ab896
git push --force
```

namiltd commented:
Try patches https://github.com/namiltd/openwrt/pull/47/commits/3338c7a532b1062e5057f4b7847e3d977d828ca3
and https://github.com/namiltd/openwrt/pull/47/commits/504728dcf88ea1f55802ca762279dc7f06457943

graysky2 commented:
> Try patches https://github.com/namiltd/openwrt/commit/3338c7a532b1062e5057f4b7847e3d977d828ca3 and https://github.com/namiltd/openwrt/commit/504728dcf88ea1f55802ca762279dc7f06457943

Test building now with minimal config:
```
CONFIG_TARGET_bcm27xx=y
CONFIG_TARGET_bcm27xx_bcm2712=y
CONFIG_TARGET_bcm27xx_bcm2712_DEVICE_rpi-5=y
CONFIG_DEVEL=y
CONFIG_TOOLCHAINOPTS=y
CONFIG_BUSYBOX_CUSTOM=y
# CONFIG_BINUTILS_USE_VERSION_2_42 is not set
CONFIG_BINUTILS_USE_VERSION_2_43=y
CONFIG_BINUTILS_VERSION="2.43.1"
CONFIG_BINUTILS_VERSION_2_43=y
# CONFIG_GCC_USE_VERSION_13 is not set
CONFIG_GCC_USE_VERSION_14=y
CONFIG_GCC_VERSION="14.2.0"
CONFIG_GCC_VERSION_14=y
CONFIG_LINUX_6_12=y
CONFIG_TARGET_KERNEL_PARTSIZE=32
CONFIG_TARGET_ROOTFS_PARTSIZE=512
CONFIG_TESTING_KERNEL=y
CONFIG_KERNEL_WERROR=y
```

graysky2 commented:
Made it further but build still fails:
```
#
# configuration written to .config
#
Building backport-include/backport/autoconf.h ... done.
  CC [M]  /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.o
/scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c:1187:19: error: 'no_llseek' undeclared here (not in a function); did you mean 'noop_llseek'?
 1187 |         .llseek = no_llseek,
      |                   ^~~~~~~~~
      |                   noop_llseek
make[12]: *** [scripts/Makefile.build:244: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.o] Error 1
make[11]: *** [scripts/Makefile.build:485: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom/brcm80211/brcmfmac] Error 2
make[10]: *** [scripts/Makefile.build:485: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom/brcm80211] Error 2
make[9]: *** [scripts/Makefile.build:485: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless/broadcom] Error 2
make[8]: *** [scripts/Makefile.build:485: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/drivers/net/wireless] Error 2
make[7]: *** [Makefile:1936: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2] Error 2
make[6]: *** [Makefile.build:13: modules] Error 2
make[5]: *** [Makefile.real:105: modules] Error 2
make[4]: *** [Makefile:121: modules] Error 2
make[3]: *** [Makefile:402: /scratch/union/build_dir/target-aarch64_cortex-a76_musl/linux-bcm27xx_bcm2712/mac80211-regular/backports-6.11.2/.built] Error 2
make[3]: Leaving directory '/scratch/union/package/kernel/mac80211'
time: package/kernel/mac80211/regular/compile#0.97#0.21#1.17
    ERROR: package/kernel/mac80211 failed to build (build variant: regular).
make[2]: *** [package/Makefile:185: package/kernel/mac80211/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [package/Makefile:179: /scratch/union/staging_dir/target-aarch64_cortex-a76_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:233: world] Error 2
```

namiltd commented:
Patched: 9ebe799471551cb4eec390a3ced7365835e7d3c4, try again.

graysky2 commented:
Adding that commit allowed the build to complete.  Have not had a chance to run test.

namiltd commented:
Add
```
CONFIG_ARM_PAN=y
```
to bcm2708/config-6.12

graysky2 commented:
@namiltd - see https://github.com/openwrt/openwrt/pull/17147/commits/0ae771a3a57ba0f8986a429a7b2ffcf0b3d2ccb8

namiltd commented:
Re-cherry-pick patches from https://github.com/openwrt/openwrt/pull/16547 and https://github.com/namiltd/openwrt/pull/47

namiltd commented:
Right, in your case instead of the whole https://github.com/namiltd/openwrt/pull/47 you should cherry-pick only commits 9558bae8c18a389fea48877867bee946fa5ee3ea and 0215a0db2da1c9cf1a9b6b999300cd6c234767e9.

namiltd commented:
Add to bcm2708: #CONFIG_RPMB is not set

namiltd commented:
Yes, and further checks will show whether or not further definitions are needed.

graysky2 commented:
I added the needed/missing options.  Is there an easy way to sort the file without allowing all commented lines first?

namiltd commented:
I just updated my PRs to kernel 6.12.2.

graysky2 commented:
Failures like this https://github.com/openwrt/openwrt/actions/runs/12186781692/job/33996128696?pr=17147 are expected until rpi foundation updates to 6.12.2.  Given status of this, does not make sense for me to manually rebase, just wait for them.

namiltd commented:
Constantly updating after new kernel 6.6 lub 6.12 builds is time-consuming, especially since it is not even known whether 6.12 will become LTS, 6.13 or the next version.

castillofrancodamian commented:
> Agreed. I do not think that 6.13 will be the next LTS as 6.13-rc1 was only just released and it will be a minimum of 7 rcs for it putting the final release in the Jan/2025 time frame. I believe the last release of the calendar year typically becomes the next LTS.

That's what I've been observing for the past few years. Even so, it's already [confirmed](https://www.kernel.org/category/releases.html) that 6.12 is LTS.

namiltd commented:
"Projected EOL" is not very encouraging though.

namiltd commented:
Re-cherry-pick https://github.com/namiltd/openwrt/pull/47/commits/5cd758e78e06c693868ef942c1190d15223ea265 and https://github.com/namiltd/openwrt/pull/47/commits/14da287fa6fdaf8c689862f19872f5d37b903b09

namiltd commented:
And again more changes in main so I had to rebuild my patches again. Repeat the whole operation.

namiltd commented:
Exactly. But there's no response from him, so these will have to suffice for now.

graysky2 commented:
Guessing he is busy finalizing 24.10...

namiltd commented:
Maybe backports make sense when kernel 6.13 is released and now we should use drivers directly from the kernel instead?

namiltd commented:
You can use 27d6a46f4a0e026e5c01dd41c2c049e3086d8f2f as a base, skip the x86 patches.

mantas-p committed:
generic: 6.12: add missing symbols

These symbols only become visible when some packages or kernel debug
options get enabled. Move them to generic.

Signed-off-by: Mantas Pucka <mantas@8devices.com>

mantas-p committed:
config: add new options for 6.12 kernel

Since kernel v6.12 couple more options were added to
CGROUP section. Add them here to avoid missing symbols.

Signed-off-by: Mantas Pucka <mantas@8devices.com>

namiltd commented:
Please update a7b89e638683986e54580f49930185478c877519

namiltd committed:
generic: 6.12: remove obsolete pending patches

Remove obsolete pending patches

Removed (already included):
 - 410-mtd-spinand-set-bitflip_threshold-to-75-of-ECC-strength.patch
 - 489-mtd-spinand-winbond-add-support-for-W25N01KV.patch
 - 680-net-add-TCP-fraglist-GRO-support.patch
 - 685-net-gso-fix-tcp-fraglist-segmentation-after-pull-fro.patch
 - 704-netfilter-nf_tables-fix-bidirectional-offload-regres.patch
 - 713-03-arm64-dts-qcom-ipq8074-add-clock-frequency-to-MDIO-n.patch
 - 735-net-ethernet-mtk_eth_soc-fix-memory-corruption-durin.patch
 - 736-net-ethernet-mtk_wed-fix-path-of-MT7988-WO-firmware.patch
 - 739-02-phy-add-driver-for-MediaTek-XFI-T-PHY.patch
 - 880-01-dt-bindings-leds-add-LED_FUNCTION_MOBILE-for-mobile-.patch
 - 880-02-dt-bindings-leds-add-LED_FUNCTION_SPEED_-for-link-sp.patch
 - 980-tools-thermal-tmon-Fix-compilation-warning-for-wrong.patch

Temporary removed (for further investigation):
 - 305-mips_module_reloc.patch
 - 811-pci_disable_usb_common_quirks.patch
 - 820-w1-gpio-fix-problem-with-platfom-data-in-w1-gpio.patch

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

mantas-p committed:
generic: 6.12: remove outdated kernel symbols

Symbols that no longer exist in 6.12 were removed by a script [1].

[1] https://github.com/openwrt/openwrt/pull/15324

Signed-off-by: Mantas Pucka <mantas@8devices.com>

namiltd commented:
In kernel 6.6.64 the following files have changed significantly:
	bcm27xx/patches-6.6/950-0585-drm-vc4-Introduce-generation-number-enum.patch
	bcm27xx/patches-6.6/950-0610-drm-vc4-hvs-Support-BCM2712-HVS.patch
	bcm27xx/patches-6.6/950-0829-vc4-hvs-Add-support-for-D0-register-changes.patch

Additionally, the following have been removed:
	bcm27xx/patches-6.6/950-0597-drm-vc4-hdmi-Avoid-hang-with-debug-registers-when-su.patch
	bcm27xx/patches-6.6/950-0599-drm-vc4-Fix-dlist-debug-not-resetting-the-next-entry.patch
	bcm27xx/patches-6.6/950-0600-drm-vc4-Remove-incorrect-limit-from-hvs_dlist-debugf.patch
	bcm27xx/patches-6.6/950-0708-drm-vc4-Correct-logic-on-stopping-an-HVS-channel.patch

So Your PR needs a major "overhaul"

graysky2 commented:
That is odd, I just brought it back to match openwrt/openwrt, applied your 6.12 branch, then ran `./scripts/kernel_bump.sh -s 6.6 -t 6.12 -p bcm27xx`, applied my testing kernel commit, replaced the `patches-6.12/` with a freshly generated set, and applied my config 6.12 commit.  Why is the script touching the 6.6 patches?

namiltd commented:
kernel_bump.sh moves the patches from version 6.6 to version 6.12 and then recreates version 6.6. So you need to re-execute kernel_bump every time version 6.6 changes, because it will recreate the old patches

graysky2 commented:
I see... yes, having your 6.12 merged will simplify things.

DragonBluep commented:
@graysky2 Hi! Thanks for your contribution. Since bcm27xx SoCs are maintained by the Raspberry Pi Foundation. I would suggest backport rpi official 6.12 based patches from their repository instead of refreshing the 6.6 patches. https://github.com/raspberrypi/linux/tree/rpi-6.12.y

And it's too early to support 6.12 on openwrt. Perhaps you can temporarily pause this PR to avoid some annoying and unnecessary patch refresh work.

graysky2 commented:
@DragonBluep - this is precisely how I have generated the patches for bcm27xx included in this PR.

namiltd commented:
@DragonBluep I know, but LTS kernel versions are currently only supported for a maximum of 2 years, so time is running out.

DragonBluep commented:
> @DragonBluep - this is precisely how I have generated the patches for bcm27xx included in this PR.

Yes I see it. But I think the first two commits Create/Restore kernel files are useless. Most patch names have been changed in the new kernel. Unless you want to spend hours of time to reorganize them, it's hard to say we can track something from such a big and terrible patch. As you can see https://github.com/openwrt/openwrt/pull/17147/commits/1bf72a917ba8e81ef0373aabe80f9e776a65836a.

graysky2 commented:
>  But I think the first two commits Create/Restore kernel files are useless.

I am fine with excluding those if it's just as simple to just start at the addition of the RPi Foundation's patch set.

namiltd committed:
generic: 6.12: remove obsolete hack patches

Remove obsolete hack patches

Removed (already included):
 - 200-tools_portability.patch
 - 220-arm-gc_sections.patch
 - 721-net-add-packet-mangeling.patch
 - 790-SFP-GE-T-ignore-TX_FAULT.patch

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
generic: 6.12: remove CONFIG_DEBUG_TIMEKEEPING

In kernel 6.12.5 the CONFIG_DEBUG_TIMEKEEPING setting was removed in a commit:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.5&id=7b7a9175c63bac41b07a314f052f0716c728f31f

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

georgemoussalem committed:
mac80211: ath11k: poll reo status ring for IPQ5018

This downstream patch fixes a bug which could flood the logs with the
following message and would eventually lead to a crash.

ath11k c000000.wifi: failed to send HAL_REO_CMD_UPDATE_RX_QUEUE cmd, tid 0 (-105)

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd committed:
generic: mtdsplit: include appropriate headers for kernel 6.12

Include appropriate headers for kernel 6.12

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mtk-mmc: adding brackets to prevent warnings

Fix error: suggest braces around empty body in an 'else' statement

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
Please rebase.

graysky2 commented:
@namiltd - I am thinking to wait for your initial 6.12 PR to get merged so the base of this one isn't changing so much which will avoid work for me:

> And it's too early to support 6.12 on openwrt. Perhaps you can temporarily pause this PR to avoid some annoying and unnecessary patch refresh work.

namiltd commented:
I think that if there are no working platforms, PR for target generic will not be approved soon. So the more of them, the faster the approval will be.

namiltd committed:
kernel/hwmon-core: update for kernel 6.12

Update dependencies for kernel 6.12

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

graysky2 committed:
bcm27xx: Create 6.12 kernel config files

Ran: make kernel_menuconfig CONFIG_TARGET=target then
make kernel_menuconfig CONFIG_TARGET=subtarget then
manually added needed missing config options base on building
the build bot config.  Corrected order using
scripts/kconfig-reorder.sh

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
patches-6.12: bcm27xx: refresh patch set

The patches were generated from the RPi repo with the following command:
git format-patch -N v6.12.12..HEAD
(HEAD -> 1f193661c9a0eb916b3c64c7a5d5dba31948b0c3)

Then used update_kernel.sh to rebase adjusting as needed.

Removed patches for Github workflow, and defconfigs.
The following was used to flag potential patches to exclude:

grep -i defconfig *.patch | awk '{ print $1 }' | sed 's/:.*$//' | uniq
grep -i github *.patch
Grep ISSUE_TEMPLATE *.patch

ls | grep -i brcm
ls | grep -i github
ls | grep -i config
ls | grep -i readme

Also helpful to grep for keywords in patches to manually inspect
like: grep defconfig *.patch

Signed-off-by: John Audia <therealgraysky@proton.me>

namiltd committed:
kernel/generic: update config for kernel 6.12

Update generic config for kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: update config for kernel 6.12

Update target x86 config for kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
package/zram: update for kernel 6.12

Updated for compatibility with kernels 6.6 and 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mac80211: temporary patch for kernel 6.12 (version 6)

Temporary patch for kernel 6.12

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel: add define for kernel 6.12

Add define for kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

aiamadeus committed:
tools: add xxhash package for ccache

The ccache build requires xxhash, which is download from
the Internet. So add host-build package for xxhash.

Fixes: #16851
Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
[ use git source proto ]
Link: https://github.com/openwrt/openwrt/pull/17700
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

aiamadeus committed:
tools: ccache: disable build testing

The ccache build relies on doctest, which
is useless and can be disabled.

Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
Link: https://github.com/openwrt/openwrt/pull/17700
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

namiltd committed:
generic: 6.12: manually rebuilding hack patches

Manually rebuilding hack patches:
 - 204-module_strip.patch
 - 210-darwin_scripts_include.patch
 - 251-kconfig.patch
 - 421-drivers-mtd-parsers-add-nvmem-support-to-cmdlinepart.patch
 - 610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch
 - 725-net-phy-aquantia-add-PHY_IDs-for-AQR112-variants.patch
 - 760-net-usb-r8152-add-LED-configuration-from-OF.patch
 - 800-GPIO-add-named-gpio-exports.patch
 - 901-debloat_sock_diag.patch
 - 902-debloat_proc.patch
 - 910-kobject_uevent.patch
 - 911-kobject_add_broadcast_uevent.patch
 - 930-Revert-Revert-Revert-driver-core-Set-fw_devlink-on-b.patch

All other patches automatically rebased.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

svanheule committed:
realtek: drop old RTL8231 driver

The old RTL8231 driver integrated the MDIO bus access with the GPIO
control ops, making this driver not very portable to newer platforms.
It depended on the SoC ID instead of the compatible to determine the
MDIO access register, further complicating portability.

A new MFD driver is now available, which offers proper pin config as
well as optional LED support, which can work on any (bitbanged) MDIO
bus. Now that all devices have been migrated, we can drop the old code.

Signed-off-by: Sander Vanheule <sander@svanheule.net>

namiltd committed:
generic: 6.12: remove obsolete backport patches

Remove obsolete backport patches already included in kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
generic: 6.12: manually rebuilding backport patches

Manually rebuilding backport patches:
 - 410-v6.13-01-block-add-support-for-defining-read-only-partitions.patch
 - 839-v6.13-net-phy-aquantia-allow-forcing-order-of-MDI-pairs.patch
 - 901-v6.13-net-dsa-mv88e6xxx-Support-LED-control.patch

All other patches automatically rebased.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

graysky2 committed:
kernel: bump 6.6 to 6.6.75

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.75

All patches automatically rebased.

*** WARNING - the image I built/flashed to my RPi5B, failed to boot!
...
block: attempting to load /tmp/overlay/upper/etc/config/fstab
block: unable to load configuration (fstab: Entry not found)
block: attempting to load /tmp/overlay/etc/config/fstab
block: unable to load configuration (fstab: Entry not found)
block: attempting to load /etc/config/fstab
block: unable to load configuration (fstab: Entry not found)
block: no usable configuration

I am not sure if this is due to changes in OpenWrt's code since I build
the last-known-good image (r28670-0fd9d00cd6fc) or if this is something
in the 6.6.75 kernel bump.

Build system: x86/64
Build-tested: bcm27xx/bcm2712
Run-tested: FAILED TO BOOT

Signed-off-by: John Audia <therealgraysky@proton.me>

csharper2005 committed:
kernel: nmbm: add mediatek,bmt-mtd-overridden-oobsize property

This commit adds new "mediatek,bmt-mtd-overridden-oobsize" property. The
property helps avoid "NMBM configuration mismatch" error if mtd "OOB size"
is not equal to the "spare size" which is stored in the nmbm signature.

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17549
Signed-off-by: Robert Marko <robimarko@gmail.com>

csharper2005 committed:
mediatek: fix nmbm configuration mismatch (Xiaomi AX3000t)

This commit fixes nmbm configuration mismatch error on Xiaomi AX3000t
with Winbond W25N01KVZEIR spi-nand:
'''
[    0.786783] NMBM configuration mismatch
'''

Root cause:
1. U-Boot W25N01KV spi-nand driver ia compiled with 64B OOB size for the
   chip and store this size in the nmbm signature;
2. Linux W25N01KV driver use 96B OOB.

The change doesn't affect AX3000t variants with other spi-nand chips
(ESMT, Foresee) because their Linux drivers use 64B OOB.

Fixes: openwrt#16972
Tested-by: Aleksandr Danilov <sc16me@gmail.com>
Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17549
Signed-off-by: Robert Marko <robimarko@gmail.com>

wigyori committed:
imx-bootlets: bump to 10.12.01

 - refresh patches
 - add patch to fix compilation with GCC13
   (referring to ARM-8933/1).

Link: https://lore.kernel.org/lkml/20220630133231.272605764@linuxfoundation.org/
Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

namiltd committed:
kernel/generic: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/generic: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
generic: 6.12: manually rebuilding pending patches

Manually rebuilding pending patches:
 - 103-kbuild-export-SUBARCH.patch
 - 141-jffs2-add-RENAME_EXCHANGE-support.patch
 - 203-kallsyms_uncompressed.patch
 - 270-platform-mikrotik-build-bits.patch
 - 308-mips32r2_tune.patch
 - 330-MIPS-kexec-Accept-command-line-parameters-from-users.patch
 - 402-mtd-spi-nor-write-support-for-minor-aligned-partitions.patch
 - 451-block-partitions-populate-fwnode.patch
 - 476-mtd-spi-nor-add-eon-en25q128.patch
 - 477-mtd-spi-nor-add-eon-en25qx128a.patch
 - 479-mtd-spi-nor-add-xtx-xt25f128b.patch
 - 481-mtd-spi-nor-add-support-for-Gigadevice-GD25D05.patch
 - 482-mtd-spi-nor-add-gd25q512.patch
 - 484-mtd-spi-nor-add-esmt-f25l16pa.patch
 - 485-mtd-spi-nor-add-xmc-xm25qh128c.patch
 - 488-mtd-spi-nor-add-xmc-xm25qh64c.patch
 - 490-ubi-auto-attach-mtd-device-named-ubi-or-data-on-boot.patch
 - 497-mtd-mtdconcat-add-dt-driver-for-concat-devices.patch
 - 498-mtd-spi-nor-locking-support-for-MX25L6405D.patch
 - 510-block-add-uImage.FIT-subimage-block-driver.patch
 - 530-jffs2_make_lzma_available.patch
 - 630-packet_socket_type.patch
 - 666-Add-support-for-MAP-E-FMRs-mesh-mode.patch
 - 681-net-remove-NETIF_F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch
 - 700-netfilter-nft_flow_offload-handle-netdevice-events-f.patch
 - 702-net-ethernet-mtk_eth_soc-enable-threaded-NAPI.patch
 - 706-net-phy-populate-host_interfaces-when-attaching-PHY.patch
 - 711-03-net-dsa-qca8k-add-support-for-port_change_master.patch
 - 737-net-ethernet-mtk_eth_soc-add-paths-and-SerDes-modes-.patch
 - 739-03-net-pcs-pcs-mtk-lynxi-add-platform-driver-for-MT7988.patch
 - 739-05-net-pcs-add-driver-for-MediaTek-USXGMII-PCS.patch
 - 742-net-phy-air_en8811h-reset-netdev-rules-when-LED-is-s.patch
 - 801-gpio-gpio-cascade-add-generic-GPIO-cascade.patch
 - 809-01-nvmem-core-generalize-mac-base-cells-handling.patch
 - 834-ledtrig-libata.patch
 - 870-ARM-dts-nxp-imx7d-pico-add-cpu-supply-nodes.patch
 - 892-leds-Add-LED1202-I2C-driver.patch
 - 920-mangle_bootargs.patch

All other patches automatically rebased.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

robimarko committed:
ipq-wifi: update to Git HEAD (2025-02-06)

12948ce23744 Revert "ipq6018: add BDF for Edgecore EAP101"
6b3522faae1a ipq6018: add BDF for Edgecore EAP101
0411c6b1a00a ipq5018: add BDF for Linksys MX2000
feb4f2dfaa1b qcn6122: add BDF for Linksys MX2000
29819c35c8a5 ipq5018: add BDF for Linksys MX5500
38b0e9ee0f3c qcn9074: add BDF for Linksys MX5500

Link: https://github.com/openwrt/openwrt/pull/17877
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
libiwinfo: update to Git HEAD (2025-02-06)

9ff15f7ee3a0 devices: add device id for MediaTek MT7992E
94b3a3c1a6c4 devices: add device id for Qualcomm Atheros IPQ5018
9cec6b4dd2df devices: add device id for Qualcomm Atheros QCN6122

Link: https://github.com/openwrt/openwrt/pull/17878
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
qualcommax: disable IPQ5018 specific symbols

Newly introduced IPQ5018 specific symbols will stop compilation of
ipq807x and ipq60xx subtargets, so make sure to disable them in the
generic config.

Link: https://github.com/openwrt/openwrt/pull/17884
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
qualcommax: ipq50xx: refresh kernel config

Refresh the ipq50xx kernel config as it looks like it was created by hand
instead of kernel_menuconfig so we have stuff like DWC3 which is packaged
as a kmod in a generic way, QMI helpers are part of backports etc.

Link: https://github.com/openwrt/openwrt/pull/17890
Signed-off-by: Robert Marko <robimarko@gmail.com>

nbd168 committed:
build: conditionally build llvm bpf toolchain by default

Unless another toolchain is present (or selected), build the bpf toolchain
whenever a package is selected that needs it.

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Noltari commented:
Superseded by https://github.com/openwrt/openwrt/pull/18736
Thanks @graysky2!
